packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart

```diff
@@ -106,6 +106,7 @@ class PerfettoControllerImpl extends PerfettoController {
       ..height = '100%'
       ..width = '100%';
 
+    // TODO(jacobr): why does this have to be ignored?
```

Comment:
I'm not sure on the details of why, but this is a weird flutter web thing. @ditman do you know the reasoning for this?

Reply:
This needs to be ignored because the API surface of `dart:ui` for the web is bigger than for mobile (it has the `webOnlyXXX` methods that are not defined in the mobile version, for example, or the `platformViewRegistry` object).

`analyzer` flags this as you calling "things that don't exist" in the API because it only sees the proper `dart:ui` mobile package, not the web re-implementation with extra methods.

A workaround to this ignore is to apply a _shim_ for the web version of dart:ui (but I've only used it in web-only packages), like [this one](https://github.com/flutter/plugins/tree/fd2841fd065f07ad429d56cd54ec9723f4d1953b/packages/video_player/video_player_web/lib/src/shims), [used here](https://github.com/flutter/plugins/blob/fd2841fd065f07ad429d56cd54ec9723f4d1953b/packages/video_player/video_player_web/lib/video_player_web.dart#L12).

* This is the (venerable) tracking issue: https://github.com/flutter/flutter/issues/41563

Reply:
Thanks. Referenced the issue.